src/PlexRequests.Helpers/PlexRequestException.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Net;
+
+namespace PlexRequests.Helpers
+{
+    public class PlexRequestException : Exception
+    {
+        public HttpStatusCode StatusCode { get; }
+        public string Description { get; }
+        public object LoggableObject { get; }
+
+        public PlexRequestException(string message, string description = null, HttpStatusCode statusCode = HttpStatusCode.BadRequest, object loggableObject = null) : base(message)
```

Comment:
I would define an Interface for the LoggableObject or something. Currently everything is an object, so everything is loggable?

Reply:
I should be able to pass any object and have it serialised for logging purposes. So object is correct in this instance.